inc/clr/prof/corprof.h

```diff
@@ -14868,537 +14876,1274 @@ EXTERN_C const IID IID_ICorProfilerInfo9;
     ( (This)->lpVtbl -> GetInMemorySymbolsLength(This,moduleId,pCountSymbolBytes) ) 
 
-#define ICorProfilerInfo9_ReadInMemorySymbols(This,moduleId,symbolsReadOffset,pSymbolBytes,countSymbolBytes,pCountSymbolBytesRead)	\
+#define ICorProfilerInfo9_ReadInMemorySymbols(This,moduleId,symbolsReadOffset,pSymbolBytes,countSymbolBytes,pCountSymbolBytesRead)  \
     ( (This)->lpVtbl -> ReadInMemorySymbols(This,moduleId,symbolsReadOffset,pSymbolBytes,countSymbolBytes,pCountSymbolBytesRead) ) 
 
 
-#define ICorProfilerInfo9_IsFunctionDynamic(This,functionId,isDynamic)	\
+#define ICorProfilerInfo9_IsFunctionDynamic(This,functionId,isDynamic)  \
     ( (This)->lpVtbl -> IsFunctionDynamic(This,functionId,isDynamic) ) 
 
-#define ICorProfilerInfo9_GetFunctionFromIP3(This,ip,functionId,pReJitId)	\
+#define ICorProfilerInfo9_GetFunctionFromIP3(This,ip,functionId,pReJitId)   \
     ( (This)->lpVtbl -> GetFunctionFromIP3(This,ip,functionId,pReJitId) ) 
 
-#define ICorProfilerInfo9_GetDynamicFunctionInfo(This,functionId,moduleId,ppvSig,pbSig,cchName,pcchName,wszName)	\
+#define ICorProfilerInfo9_GetDynamicFunctionInfo(This,functionId,moduleId,ppvSig,pbSig,cchName,pcchName,wszName)    \
     ( (This)->lpVtbl -> GetDynamicFunctionInfo(This,functionId,moduleId,ppvSig,pbSig,cchName,pcchName,wszName) ) 
 
 
-#define ICorProfilerInfo9_GetNativeCodeStartAddresses(This,functionID,reJitId,cCodeStartAddresses,pcCodeStartAddresses,codeStartAddresses)	\
+#define ICorProfilerInfo9_GetNativeCodeStartAddresses(This,functionID,reJitId,cCodeStartAddresses,pcCodeStartAddresses,codeStartAddresses)  \
     ( (This)->lpVtbl -> GetNativeCodeStartAddresses(This,functionID,reJitId,cCodeStartAddresses,pcCodeStartAddresses,codeStartAddresses) ) 
 
-#define ICorProfilerInfo9_GetILToNativeMapping3(This,pNativeCodeStartAddress,cMap,pcMap,map)	\
+#define ICorProfilerInfo9_GetILToNativeMapping3(This,pNativeCodeStartAddress,cMap,pcMap,map)    \
     ( (This)->lpVtbl -> GetILToNativeMapping3(This,pNativeCodeStartAddress,cMap,pcMap,map) ) 
 
-#define ICorProfilerInfo9_GetCodeInfo4(This,pNativeCodeStartAddress,cCodeInfos,pcCodeInfos,codeInfos)	\
+#define ICorProfilerInfo9_GetCodeInfo4(This,pNativeCodeStartAddress,cCodeInfos,pcCodeInfos,codeInfos)   \
     ( (This)->lpVtbl -> GetCodeInfo4(This,pNativeCodeStartAddress,cCodeInfos,pcCodeInfos,codeInfos) ) 
 
 #endif /* COBJMACROS */
 
 
-#endif 	/* C style interface */
+#endif  /* C style interface */
 
 
 
 
-#endif 	/* __ICorProfilerInfo9_INTERFACE_DEFINED__ */
+#endif  /* __ICorProfilerInfo9_INTERFACE_DEFINED__ */
 
 
-#ifndef __ICorProfilerMethodEnum_INTERFACE_DEFINED__
-#define __ICorProfilerMethodEnum_INTERFACE_DEFINED__
+#ifndef __ICorProfilerInfo10_INTERFACE_DEFINED__
+#define __ICorProfilerInfo10_INTERFACE_DEFINED__
 
-/* interface ICorProfilerMethodEnum */
+/* interface ICorProfilerInfo10 */
 /* [local][unique][uuid][object] */ 
 
 
-EXTERN_C const IID IID_ICorProfilerMethodEnum;
+EXTERN_C const IID IID_ICorProfilerInfo10;
```

Comment:
>EXTERN_C const IID IID_ICorProfilerInfo10 [](start = 0, length = 41)

Are we updating the version of the PAL as well? If we aren't then I'm unsure whether or not IID_ICorProfilerInfo10 will be linked into the final .a file, which may cause undefined behavior.

Reply:
The latest PAL does not have ICorProfilerInfo10 defined. I don't think we are using this symbol anywhere, so what I'm thinking is that I can try to undefine it, make it a compiler error if used (not sure if that it possible), or redefine it with another header file (we know what the value should be). I'm thinking the last option.

Reply:
We may or may not be using the symbol. It depends on the way that uuidof is implemented in our version of clang. I think some older versions just defined it as:

#define uuidof(interface) IID_##interface

so, uuidof(ICorProfilerInfo10) would just resolve to IID_ICorProfilerInfo10.

You can't undefine a symbol, unfortunately. The problem will be with the linker. If there is no value set for the extern, then there will just be a random value put in place of the address of the const variable, and your behavior is undefined. The most likely thing is that the QI will just fail because you'll have a garbage GUID, but that is no guarantee.

You could just put it in our own cpp file. I think the linker will complain if there is an attempt to link the same symbol more than once.

Reply:
If I don't define it and we try to use it in our code, then we'll get a linker error when building on Windows stating "error LNK2001: unresolved external symbol IID_ICorProfilerInfo10"; on Linux, there will not be an error because of the delayed symbol resolution.

If I do define it, there's nothing that is using it, so I don't have anything that actually verifies that the definition satisfies the extern declaration. Ignoring that, I'm not sure I can put it in a shared cpp file since I have to reference stdafx.h, which is in a different location for each project. Creating another library project just for this seems a little overkill.

I'm reasonably certain that using CComQIPtr<ICorProfilerInfo10> or QueryInterface with __uuidof(ICorProfilerInfo10) works without this being defined for both Windows and Linux (I'll reconfirm that).

I'm leaning toward not defining this since we aren't using it and we would very likely get a linker error when we do start using it.

Reply:
We are only not using the symbol because we are using uuidof. If clang actually looks at the declspec in our version, then we will continue to not use it. But, if it doesn't, then we are already using it in the Linux build in CorProfilerInfoWrapper.cpp. That's where I'm worried about the undefined behavior.

Reply:
After some discussion... it seems that clang has support for __uuidof? I can't find anything in the official docs about it, though. I'll sign off.